connectors/sources/confluence.py

```diff
@@ -256,6 +323,214 @@ async def ping(self):
+
+    async def fetch_attachments(
+        self, content_id, parent_name, parent_space, parent_type
+    ):
+        """Fetches all the attachments present in the given content (pages and blog posts)
+
+        Args:
+            content_id (str): Unique identifier of Confluence content (pages and blogposts)
+            parent_name (str): Page/Blog post where the attachment is present
+            parent_space (str): Confluence Space where the attachment is present
+            parent_type (str): Type of the parent (Blog post or Page)
+
+        Yields:
+            Dictionary: Confluence attachment on the given page or blog post
+            String: Download link to get the content of the attachment
+        """
+        async for response in self.confluence_client.paginated_api_call(
+            url_name=ATTACHMENT,
+            api_query=ATTACHMENT_QUERY,
+            id=content_id,
+        ):
+            attachments = response.get("results", [])
+            for attachment in attachments:
+                yield {
+                    "type": attachment["type"],
+                    "title": attachment["title"],
+                    "_id": attachment["id"],
+                    "space": parent_space,
+                    parent_type: parent_name,
+                    "_timestamp": attachment["version"]["when"],
+                    "size": attachment["extensions"]["fileSize"],
+                    "url": os.path.join(
+                        self.confluence_client.host_url,
+                        attachment["_links"]["webui"][1:],
+                    ),
+                }, attachment["_links"]["download"]
+
+    async def download_attachment(self, url, attachment, timestamp=None, doit=False):
+        """Downloads the content of the given attachment in chunks using REST API call
+
+        Args:
+            url (str): url endpoint to download the attachment content
+            attachment (dict): Dictionary containing details of the attachment
+            timestamp (timestamp, optional): Timestamp of Confluence last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to False.
+
+        Returns:
+            Dictionary: Document of the attachment to be indexed.
+        """
+        attachment_size = int(attachment["size"])
+        if not (self.enable_content_extraction and doit and attachment_size):
+            return
+        attachment_name = attachment["title"]
+        if os.path.splitext(attachment_name)[-1] not in TIKA_SUPPORTED_FILETYPES:
+            logger.warning(f"{attachment_name} can't be extracted")
+            return
+
+        if attachment_size > FILE_SIZE_LIMIT:
+            logger.warning(
+                f"File size {attachment_size} of file {attachment_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding file content"
+            )
+            return
+        logger.debug(f"Downloading {attachment_name}")
```

Comment:
Would be great if we can have the size information here and also that it's downloaded from confluence to improve traceability.